src/Lecture2.hs

```diff
@@ -52,7 +58,11 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct l =
+  if elem 0 l == True
+    then 0
+    else product l
```

Comment:
That's indeed a very lazy solution 😅 
Not the one I expected but whatever works 🙂